arangod/Pregel/StatusWriter/CollectionStatusWriter.h

```diff
@@ -58,6 +58,17 @@ struct CollectionStatusWriter : StatusWriterInterface {
     _logicalCollection = std::move(logicalCollection);
   };
 
+  CollectionStatusWriter(TRI_vocbase_t& vocbase) : _vocbaseGuard(vocbase) {
+    CollectionNameResolver resolver(_vocbaseGuard.database());
+    auto logicalCollection =
+        resolver.getCollection(StaticStrings::PregelCollection);
+    if (logicalCollection == nullptr) {
+      THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND,
+                                     StaticStrings::PregelCollection);
+    }
+    _logicalCollection = std::move(logicalCollection);
+  };
+
```

Comment:
As this header file does now include way too much code - we need to split this up into .cpp and .h files.

Reply:
https://arangodb.atlassian.net/browse/GORDO-1592